mailbox/event/json/src/test/java/org/apache/james/event/json/MailboxACLUpdatedEventSerializationTest.java

```diff
@@ -0,0 +1,572 @@
+                    "   }" +
+                    "}").get())
+                    .isInstanceOf(NoSuchElementException.class);
+            }
+        }
+
+        @Nested
+        class DeserializationErrorOnUser {
+
+            @Test
+            void mailboxACLUpdatedShouldThrowWhenMissingUser() {
+                assertThatThrownBy(() -> EVENT_SERIALIZER.fromJson(
+                    "{" +
+                    "  \"MailboxACLUpdated\":{" +
+                    "    \"mailboxPath\":{" +
+                    "       \"namespace\":\"#private\"," +
+                    "       \"name\":\"mailboxName\"" +
+                    "      }," +
+                    "    \"aclDiff\":{" +
+                    "       \"oldACL\":{\"entries\":{}}," +
+                    "       \"newACL\":{\"entries\":{\"$any\":\"ar\"}}}," +
+                    "    \"mailboxId\":\"23\"," +
+                    "    \"sessionId\":6" +
+                    "   }" +
+                    "}").get())
+                    .isInstanceOf(NoSuchElementException.class);
+            }
+
+            @Test
+            void mailboxACLUpdatedShouldThrowWhenUserIsNotAString() {
+                assertThatThrownBy(() -> EVENT_SERIALIZER.fromJson(
+                    "{" +
+                    "  \"MailboxACLUpdated\":{" +
+                    "    \"mailboxPath\":{" +
+                    "       \"namespace\":\"#private\"," +
+                    "       \"name\":\"mailboxName\"" +
+                    "      }," +
+                    "    \"aclDiff\":{" +
+                    "       \"oldACL\":{\"entries\":{}}," +
+                    "       \"newACL\":{\"entries\":{\"$any\":\"ar\"}}}," +
+                    "    \"mailboxId\":\"23\"," +
+                    "    \"sessionId\":6," +
+                    "    \"user\":12345" +
+                    "   }" +
+                    "}").get())
+                    .isInstanceOf(NoSuchElementException.class);
+            }
+
+            @Test
+            void mailboxACLUpdatedShouldThrowWhenUserIsNotWellFormatted() {
+                assertThatThrownBy(() -> EVENT_SERIALIZER.fromJson(
+                    "{" +
+                    "  \"MailboxACLUpdated\":{" +
+                    "    \"mailboxPath\":{" +
+                    "       \"namespace\":\"#private\"," +
+                    "       \"name\":\"mailboxName\"" +
+                    "      }," +
+                    "    \"aclDiff\":{" +
+                    "       \"oldACL\":{\"entries\":{}}," +
+                    "       \"newACL\":{\"entries\":{\"$any\":\"ar\"}}}," +
+                    "    \"mailboxId\":\"23\"," +
+                    "    \"sessionId\":6," +
+                    "    \"user\":\"user@domain@secondDomain\"" +
+                    "   }" +
+                    "}").get())
+                    .isInstanceOf(IllegalArgumentException.class);
+            }
+        }
+
+        @Nested
+        class DeserializationErrorOnACLDiff {
```

Comment:
I believe that we are missing some tests here:

 - what happens when oldACL is null? is a String? Contains some AclEntryKey that can not be deserialized (empty string? Or `-`)? When rights is null? Not a String? Invalid (like `\"invalid\"`)? Have double right entry `\"rr\"` ?
 - Same thing for new ACL

ACL is a complex object, and we should try to cover the edge cases with tests...